pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java

```diff
@@ -83,6 +85,27 @@ public static List<StarTreeV2BuilderConfig> generateBuilderConfigs(@Nullable Lis
     return builderConfigs;
   }
 
+  public static List<StarTreeV2BuilderConfig> generateBuilderConfigs(@Nullable List<StarTreeIndexConfig> indexConfigs,
+      boolean enableDefaultStarTree, Schema schema, JsonNode segmentMetadata) {
+    List<StarTreeV2BuilderConfig> builderConfigs = new ArrayList<>();
+    if (indexConfigs != null) {
+      for (StarTreeIndexConfig indexConfig : indexConfigs) {
+        StarTreeV2BuilderConfig builderConfig = StarTreeV2BuilderConfig.fromIndexConfig(indexConfig);
+        if (!builderConfigs.contains(builderConfig)) {
```

Comment:
`StarTreeV2BuilderConfig` <- can we double check the `equals()` function for this class?

Reply:
`equals()` is properly overridden properly : [here](https://github.com/apache/pinot/blob/5c81c0a88a0a95e5bf150363bb3ed5b5203de795/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeV2BuilderConfig.java#L245-L256)